tools/src/lib/tokensSearch.ts

```diff
@@ -0,0 +1,78 @@
+import { tool } from '@langchain/core/tools';
+import { z } from 'zod';
+import axios from 'axios';
+import qs from 'qs';
+import { PortalsResponse, TokenSearchResult } from './types';
+
+export const tokensSearch = tool(
```

Comment:
From a developer experience perspective we might want to use a pattern that separated the function param from the fields param, e.g.

```ts
const fn = ...
const fields = ...
export const tokenSearch = tool(fn, fields)
```

Reply:
Happy to tackle in the follow-up PR as a few things change there tools-wise!